src/components/dialog.js

```diff
@@ -261,17 +264,46 @@ export class Dialog {
     this.getContainer().appendChild(view.getElement());
 
     // If the current view should fade the parent document.
-    if (view.shouldFadeBody()) {
+    if (view.shouldFadeBody() && !this.hidden_) {
       this.graypane_.show(/* animate */ true);
     }
+
     return view.init(this).then(() => {
       setImportantStyles(view.getElement(), {
         'opacity': 1,
       });
+      if (this.hidden_ && view.shouldFadeBody()) {
+        this.graypane_.show(/* animated */ true);
+      }
+      this.show_();
+      this.hidden_ = false;
       this.setLoading(false);
     });
   }
 
+  /**
+   * Show the iframe
+   * @private
+   * @param {boolean=} showNow
+   */
+  show_(showNow = this.hidden_) {
```

Comment:
What does `showNow` really mean? It feels to me that by extension, this method should only have `animated` boolean param.

Reply:
addressed.